app/books/routers/routers.py

```
@@ -0,0 +1,78 @@
+from utils.exceptions.exception import BookDoesNotExist, BackEndError
+from app.books.service.service import BookService
+from app.books.schema.schema import BookSchema
+from utils.responses.response import Response
+from flask.views import MethodView
+from flask_classy import FlaskView
+from flask import request, jsonify
+from functools import wraps
+from const import data_help
+
+import json
+
+schema = BookSchema()
+response = Response()
+book_service = BookService()
+
+
+def schema_load(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        data = schema.loads(request.data)
+        kwargs['data'] = data
+        result = func(*args, **kwargs)
+        return result
+    return wrapper
+
+
+class HelpView(FlaskView, MethodView):
+    def get(self):
+        return json.dumps(data_help)
+
+
+class BookView(FlaskView):
+    def index(self):
+        result = None
+        try:
+            result = book_service.list()
+        except BackEndError:
+            return response.response_503()
+        result = schema.dump(result, many=True)
+        return jsonify(result.data)
+
+    def get(self, book_id):
+        result = None
+        try:
+            result = book_service.one(book_id)
+        except BookDoesNotExist:
+            return response.response_404()
+        except BackEndError:
+            return response.response_503()
+        result = schema.dump(result)
+        return jsonify(result.data)
+
+    @schema_load
+    def post(self, **kwargs):
+        data = kwargs['data']
+        if data.errors:
```

1. you can raise an exception in case of **marshmallow** errors in `schema_load` (instead of here)
1. you can intercept the raised exception in one place at app.handler level (instead of doing this in every view)